src/core/tests/type_prop/col2im.cpp

```diff
@@ -0,0 +1,215 @@
+    const auto kernel_size = std::make_shared<ov::op::v0::Constant>(element::i64, Shape{2}, std::vector<int32_t>{2, 2});
+    const auto strides = Strides{2, 2};
+    const auto dilations = Strides{2, 2};
+    const auto pads_begin = Shape{3, 3};
+    const auto pads_end = Shape{3, 3};
+
+    OV_EXPECT_THROW(std::ignore = make_op(data, output_size, kernel_size, strides, dilations, pads_begin, pads_end),
+                    ov::NodeValidationFailure,
+                    HasSubstr("First non-batch dimension is not evenly divisible by Product(kernel_shape)"));
+}
+
+TEST_F(TypePropCol2ImTest, dynamic_input_shapes) {
+    const auto data = std::make_shared<Parameter>(element::i64, PartialShape::dynamic());
+    const auto output_size = std::make_shared<Parameter>(element::i64, PartialShape::dynamic());
+    const auto kernel_size = std::make_shared<Parameter>(element::i64, PartialShape::dynamic());
+    const auto strides = Strides{2, 2};
+    const auto dilations = Strides{2, 2};
+    const auto pads_begin = Shape{3, 3};
+    const auto pads_end = Shape{3, 3};
+
+    const auto op =
+        std::make_shared<ov::op::v15::Col2Im>(data, output_size, kernel_size, strides, dilations, pads_begin, pads_end);
+    op->validate_and_infer_types();
+
+    EXPECT_EQ(op->get_output_size(), 1);
+    EXPECT_EQ(op->get_input_size(), 3);
+    EXPECT_EQ(op->get_output_element_type(0), element::i64);
+    EXPECT_EQ(op->get_output_partial_shape(0),
+              (PartialShape{Dimension::dynamic(), Dimension::dynamic(), Dimension::dynamic(), Dimension::dynamic()}));
+}
+
+TEST_F(TypePropCol2ImTest, static_batch) {
+    const auto data =
+        std::make_shared<Parameter>(element::i64, PartialShape{5, Dimension::dynamic(), Dimension::dynamic()});
+    const auto output_size = std::make_shared<Parameter>(element::i64, PartialShape::dynamic());
+    const auto kernel_size = std::make_shared<Parameter>(element::i64, PartialShape::dynamic());
+    const auto strides = Strides{2, 2};
+    const auto dilations = Strides{2, 2};
+    const auto pads_begin = Shape{3, 3};
+    const auto pads_end = Shape{3, 3};
+
+    const auto op =
+        std::make_shared<ov::op::v15::Col2Im>(data, output_size, kernel_size, strides, dilations, pads_begin, pads_end);
+    op->validate_and_infer_types();
+
+    EXPECT_EQ(op->get_output_size(), 1);
+    EXPECT_EQ(op->get_input_size(), 3);
+    EXPECT_EQ(op->get_output_element_type(0), element::i64);
+    EXPECT_EQ(op->get_output_partial_shape(0),
+              (PartialShape{5, Dimension::dynamic(), Dimension::dynamic(), Dimension::dynamic()}));
+}
+
+TEST_F(TypePropCol2ImTest, 2D_dynamic_input) {
+    const auto data =
+        std::make_shared<Parameter>(element::i64, PartialShape{Dimension::dynamic(), Dimension::dynamic()});
+    const auto output_size = std::make_shared<Parameter>(element::i64, PartialShape::dynamic());
+    const auto kernel_size = std::make_shared<Parameter>(element::i64, PartialShape::dynamic());
```

Comment:
Let's add a test where `output_size` and `kernel_size` are provided as a result of `ShapeOf` operation.
Also please add some checks for "shape symbols" in the tests.
Example:
https://github.com/openvinotoolkit/openvino/blob/0c27ee5dd83df3a1b1fcde2d4b597fff46c99286/src/core/tests/type_prop/eye.cpp#L99-L109

Reply:
Both done.